pkg/pgn/types.go

```diff
@@ -57,6 +58,7 @@ type PGNParser struct {
 	clock      time.Time
 	file       *os.File
 	bzipReader *bzip2.Reader
+	zstReader  *zstd.Decoder
```

Comment:
bzip2.Reader and zstd.Decoder are both implementing the io.Reader interface and it would be best to combine the two fields under the a single Reader interface.

That would require more changes and the use of type switching and abstracting the direct usage of the readers